### PR TITLE
Fix bug that settings are not properly saved when app is exiting

### PR DESCRIPTION
**Pull request by CrendKing**
Discussion at https://github.com/whyboris/Video-Hub-App/issues/675.

### CODE REVIEW

**Comment by whyboris** [comment]
Thank you again for your contribution 🙇 
I've not forgotten - just haven't gotten back to VHA in a while 😓 (busy with life stuff 👌 )
I hope to find more time this winter (and hopefully earlier) 😅

**Comment by cal2195** [comment]
This seems like a pretty important fix we should probably merge - logic from the bug is sound too. 🐛

**Comment by whyboris** [comment]
I intend to clean up the PR queue before releasing the next (bugfix) release 😅 
Hoping to find time for it in the coming months 🤞

**Comment by whyboris** [comment]
ps - brief summary: I moved to a new state (in the US) and bought a house; it's been great fun doing house work (doing some electrical stuff, wood/drywall, etc; currently building a kitchen bought from IKEA but have to first upgrade the circuit breaker panel to a larger on and then run new electrical through the crawlspace). 

tl;dr - have been super-eager to get back to VHA but have been busy 😅

**Comment by whyboris** [comment]
😓 sorry I've still not gotten back to serious VHA development this year -- this winter I'll get to it 🤞 

Thank you again for your contribution 🙇

**Comment by whyboris** [comment]
🤞 seems like it should work 😅 
I'm eager to release -- so I'm 🚀 merging in 🤞
